### ISSUE TITLE
[BUG] Seduction keeping summoner in combat

### ISSUE THREAD

**Issue by XyzKangUI** [ClassicWoWCommunity/cata-classic-bugs#443]
### 1. Current Behavior
A warlock does not drop combat while his Succubus pet is channeling Seduction on an enemy. So the warlock his out of combat timer is ~14s (= succubus charm duration + combat timer after). The weird thing here is that in general normal pet attacks such as swings keep the summoner in combat, but special abilities overall, e.g. Spell lock or devour magic do not affect the summoners combat state.
  #### 1.2. How to Reproduce
1. Get in combat on warlock
2. Use Seduction with Succubus
3. Profit
  #### 1.3  Source Material
  
![seduction](https://user-images.githubusercontent.com/94811434/193493395-de6476ca-ab61-42b1-838a-fda2798375d4.gif)

#### 2.1. Description
  Pet special abillities never affect the summoners combat state. Neither should seduction in this case. 
#### 2.2 Source Material
Here the lock drops combat eventhough the target was charmed 1-2s before dropping combat. He naturally drops combat because his last hostile attack was shadowfury 5-6s prior to dropping combat:
https://youtu.be/yq3N1fP3a24&t=220

Lock drops combat 1.5s after charm gets trinketed: 
https://youtu.be/BLT8pQOZxbE?t=83

**Comment by Evolvee** [comment]
This is a global bug with all pets. As long as your pet is in combat, you cant get sapped.

E.g.: If you double fear both hunter AND his pet for 10 seconds, both units are STILL in combat even at 9 seconds of the fear.. doesnt mattet that neither unit did anything to refesh the combag, its just broken and **heavily** boosts pet comps.

This was bugged since day 1 of classic TBC. Getting sapped as a hunter was literally impossible on classic. You enter the broken invis combat first second of the arena without touching anyone and then you are in "combat" forever after... Anyone who dares to even attempt to sap a hunter gets revealed from stealth and laughed at for being "bad". Then you put both the pet and the hunter out of the game for 10 seconds with CC thinking _"oh hey surely I can resap the hunter now"_ just to find out you wasted vanish because the game is bugged. Good game design 👍

**Comment by XyzKangUI** [comment]
> 

Yes I know people abuse some exploit in arena to be permanent combat without touching anyone. But this issue is about the fact that special attacks never affecting the combat state of the summoner, while normal attacks do keep the summoner in combat.

As you see in the video, Drakedog used shadowfury on 3:40, created his portal and then used pet charm. On 3:45 he drops combat for a split second (combat indicator playerframe) eventhough he used charm right before that. This behaviour does not exist in the current wotlk classic, because charm is affecting the combat timer of the warlock. The warlocks own combat timer is "frozen" until charm wears off and then it starts when charm falls off 5.5s+ sec after that.

**Comment by XyzKangUI** [comment]
Here I re-created Drakedog's steps:
![drakedog](https://user-images.githubusercontent.com/94811434/193577702-b6a7b286-3b2b-4e6d-a42a-d0e0eedae9ef.gif)

**Comment by XyzKangUI** [comment]
In the following video the druid is charmed, he trinkets and 1.5s later the warlock leaves combat: https://youtu.be/BLT8pQOZxbE?t=83

This behaviour is unlike what is described in the other post where the dynamic combat timer starts when charm gets trinketed or ends naturally.

**Comment by JamminL** [comment]
Archiving old non-relevant issues in preparation for MOP. If I made a mistake please comment on it.
